lib/map-file-delphi.ts

```diff
@@ -65,24 +71,16 @@ export class MapFileReaderDelphi extends MapFileReader {
                 segmentLength: parseInt(matches[3], 16),
             });
         } else {
-            matches = line.match(this.regexDelphiCodeSegment);
+            matches = line.match(this.regexDelphiCodeSegment) || line.match(this.regexDelphiICodeSegment);
             if (matches) {
-                codesegmentObject = this.addressToObject(matches[1], matches[2]);
-                codesegmentObject.id = this.segments.length + 1;
-                codesegmentObject.segmentLength = parseInt(matches[3], 16);
-                codesegmentObject.unitName = matches[4];
+                const codesegmentObject = {
+                    id: this.segments.length + 1,
+                    segmentLength: parseInt(matches[3], 16),
+                    unitName: matches[4],
+                    ...this.addressToObject(matches[1], matches[2]),
+                };
 
                 this.segments.push(codesegmentObject);
-            } else {
-                matches = line.match(this.regexDelphiICodeSegment);
-                if (matches) {
-                    codesegmentObject = this.addressToObject(matches[1], matches[2]);
-                    codesegmentObject.id = this.isegments.length + 1;
-                    codesegmentObject.segmentLength = parseInt(matches[3], 16);
-                    codesegmentObject.unitName = matches[4];
-
-                    this.isegments.push(codesegmentObject);
```

Comment:
Looks like we lost the `isegment` vs `segment` distinction, am fixing in a "make tests pass" PR